File: internal/content/testdata/bench/go/subset-sum-equal-k.go
func subset_sum_to_k(nums []int, k int) bool {
	n := len(nums)
	prev := make([]bool, k+1)
	prev[0] = true
	if nums[0] <= k {
		prev[nums[0]] = true
	}
	for i := 1; i < n; i++ {
		curr := make([]bool, k+1)
		curr[0] = true
		for j := 1; j <= k; j++ {
			notTake := prev[j]
			take := false
			if nums[i] <= j {
				take = prev[j-nums[i]]
			}
			curr[j] = take || notTake
		}
		prev = curr
	}
	return prev[k]
}

File: internal/content/testdata/bench/rust/subset-sum-equal-k.rs
fn subset_sum_to_k(nums: Vec<i64>, k: i64) -> bool {
    let k = k as usize;
    let n = nums.len();
    let mut prev = vec![false; k + 1];
    prev[0] = true;
    if (nums[0] as usize) <= k {
        prev[nums[0] as usize] = true;
    }
    for i in 1..n {
        let mut curr = vec![false; k + 1];
        curr[0] = true;
        for j in 1..=k {
            let not_take = prev[j];
            let take = if (nums[i] as usize) <= j {
                prev[j - nums[i] as usize]
            } else {
                false
            };
            curr[j] = take || not_take;
        }
        prev = curr;
    }
    prev[k]
}
